rav1e_js/www/src/App.tsx

```diff
@@ -0,0 +1,118 @@
+// Copyright (c) 2020, The rav1e contributors. All rights reserved
+//
+// This source code is subject to the terms of the BSD 2 Clause License and
+// the Alliance for Open Media Patent License 1.0. If the BSD 2 Clause License
+// was not distributed with this source code in the LICENSE file, you can
+// obtain it at www.aomedia.org/license/software. If the Alliance for Open
+// Media Patent License 1.0 was not distributed with this source code in the
+// PATENTS file, you can obtain it at www.aomedia.org/license/patent.
+
+import React, { useState } from 'react';
+import { Encoder, EncoderConfig, Packet } from "rav1e";
+
+// Configuration for encoder
+const conf = new EncoderConfig()
+	.setDim(64, 96)
+	.setRdoLookaheadFrame(1);
+// JSON-string representing configuration
+const config_str = JSON.stringify(JSON.parse(conf.toJSON()), null, 4);
+
+// Encoder
+const enc = new Encoder(conf);
+
+// Frame to send to encoder
+const f = enc.newFrame();
+
+export default function App() {
+	const [encoderStatus, setEncoderStatus] = useState("");
+	const [flushing, setFlushing] = useState(false);
+	const [frameQ_length, setFrameQ_length] = useState(0);
+	const [packets, setPackets] = useState(Array<Packet>())
+
+	const sendFrame = () => {
+		enc.sendFrame(f);
+		setFrameQ_length(frameQ_length + 1);
+	};
+
+	const flush = () => {
+		enc.flush();
+		setFlushing(true);
+	}
+
+	const receivePacket = () => {
+		try {
+			const p = enc.receivePacket();
+			console.log(p.display());
+			setEncoderStatus("");
+
+			// append new packet
+			const newPackets = Array.from(packets);
+			newPackets.push(p);
+			setPackets(newPackets);
+		} catch (e) {
+			if (e === "encoded") {
+				receivePacket();
+			} else if (e === "need more data") {
+				setEncoderStatus(e + " (send more frames or flush the encoder)");
+			}
+		}
```

Comment:
Once we finish the encoding the encoder status should show completed encoding/limit reached based on `EncoderStatus::LimitReached` status.
Right now, the end-user does not know whether the encoder status have reached there or not, it is more or less a guesstimate now